### PR TITLE
rspec-puppet 2.0 updates

### PR DESCRIPTION
**Pull request by cmurphy**

### CODE REVIEW

File: manifests/setup.pp [hunk 1]
```diff
@@ -25,7 +25,7 @@
   $fragment_owner = $::osfamily ? { 'windows' => undef, default => $::id }
   $fragment_mode  = $::osfamily ? { 'windows' => undef, default => '0640' }
   # test on gid fact availability to support older facter versions
-  if defined('$gid') and $::gid and $::osfamily != 'Windows' {
+  if $::gid and $::osfamily != 'Windows' {
```
Comment (mhaskel) [inline manifests/setup.pp]: Won't this fail on old facter's without the gid fact?

File: spec/unit/defines/concat_fragment_spec.rb [hunk 2]
```diff
@@ -46,13 +46,21 @@
...
         :source  => p[:source],
         :content => p[:content],
         :alias   => "concat_fragment_#{title}",
         :backup  => false,
       })
+      # The defined() function doesn't seem to work properly with puppet 3.4 and rspec.
+      # defined() works on its own, rspec works on its own, but together they
+      # determine that $gid is not defined and cause errors here. Work around
+      # it by ignoring this check for older puppet version.
+      if Puppet::Util::Package.versioncmp(Puppet.version, '3.4.0') > 0
```
Comment (hunner) [inline spec/unit/defines/concat_fragment_spec.rb]: This should account for puppet 3.4.1 - 3.4.3 (the latest 3.4 release). Perhaps you could do `Puppet::Util::Package.versioncmp(Puppet.version, '3.5.0') < 0`

Comment (hunner) [inline spec/unit/defines/concat_fragment_spec.rb]: Or `Puppet.version.match(/^3\.4/)` ? I'm not sure if you want 3.4 or not.

**Comment by cmurphy** [comment]
@bmjen ?

**Comment by bmjen** [comment]
DOH! Apologies @cmurphy! I meant to comment before closing. Since this PR is a outdated, I was going to update and retarget your branch to 1.2.x and port the respective fixes to 2.0.x.

**Comment by cmurphy** [comment]
@bmjen gotcha, thanks!

**Pull request closed without merging**